### PR TITLE
[BEAM-1070] Call from_p12_keyfile() with the correct arguments

### PR DESCRIPTION
**Pull request by aaltay**
This code path is failing, because a wrong list of arguments is passed.
Fixing that uncovered that oauth2client depends on pyOpenSSL for this
call to work. I did not add this dependency to setup.py because, it does
not install cleanly in all environments.

As a workaround, users who would like to use this authentication method
could first do a 'pip install pyOpenSSL'. I added a test, that skips if 'pyOpenSSL' is not installed.

### CODE REVIEW

**Comment by aaltay** [comment]
R: @sb2nov

File: sdks/python/apache_beam/internal/auth_test.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  def test_create_application_client(self):
+    try:
+      test_args = [
+        'test', '--service_account_name', 'abc', '--service_account_key_file', 
+        os.path.join(
+          os.path.dirname(__file__), '..', 'tests/data/privatekey.p12')]
+      with mock.patch.object(sys, 'argv', test_args):
+        credentials = auth.get_service_credentials()
+        self.assertIsNotNone(credentials)
+    except NotImplementedError:
```
Comment (sb2nov) [inline sdks/python/apache_beam/internal/auth_test.py]: Where is this raised ?

File: sdks/python/setup.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def get_version():
     author=PACKAGE_AUTHOR,
     author_email=PACKAGE_EMAIL,
     packages=setuptools.find_packages(),
-    package_data={'apache_beam': ['**/*.pyx', '**/*.pxd']},
+    package_data={'apache_beam': ['**/*.pyx', '**/*.pxd', 'tests/data/*']},
```
Comment (sb2nov) [inline sdks/python/setup.py]: Can you document how this key was generated somewhere. May be `tests/data/README.md`

File: sdks/python/apache_beam/internal/auth_test.py (hunk 1 above)
Comment (aaltay) [inline sdks/python/apache_beam/internal/auth_test.py]: Raised in the oauth2client library, from_p12_keyfile()

https://github.com/google/oauth2client/blob/9f0618d53481a3afe2d0cf232131ba21dcf583de/oauth2client/service_account.py#L282

File: sdks/python/setup.py (hunk 2 above)
Comment (aaltay) [inline sdks/python/setup.py]: Done.

**Comment by aaltay** [comment]
Thank you, PTAL.

**Comment by sb2nov** [comment]
LGTM, Thanks.

**Comment by aaltay** [comment]
Thank you.

R: @robertwb

**Comment by aaltay** [comment]
Thank you.

**Pull request closed without merging**